tests/samples/tile_and_fuse_using_scf_for.mlir

```diff
@@ -0,0 +1,75 @@
+// RUN: iree-opt --pass-pipeline='builtin.module(hal.executable(hal.executable.variant(iree-amdaie-tile-and-fuse{tiling-level=3})))' --split-input-file %s | FileCheck %s --check-prefix=TILE-LEVEL-3
```

Comment:
Actually these are just checking each pass. These can just go into `compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/tests/` . If there are any other similar tests that were added to the samples folder please move them to the location above.